src/h5web/vis-packs/core/matrix/utils.ts

```diff
@@ -21,3 +23,29 @@ export function getFormatter(
 
   return (val) => (val as string).toString();
 }
+
+export function sliceToCsv(slice: NdArray<Primitive<PrintableType>>): string {
+  let csv = '';
+
+  if (slice.shape.length === 1) {
+    for (let i = 0; i < slice.shape[0]; i++) {
+      csv += `${slice.get(i).toString()}\n`; // complex numbers are stringifyied as two values
+    }
+
+    return csv;
```

Comment:
Alternatively, you can make use of `slice.data` for 1D slices:
```suggestion
    return slice.data.reduce((acc, val) => acc += `${val.toString()}\n`;
```

and keep `let csv = '';` inside `if (slice.shape.length === 2)` block.

Reply:
TypeScript doesn't like it, even with a type cast on `slice.data` 😢 -- maybe once the typing improvements of `ndarray` are merged...